### PR TITLE
Fix links on both github and website

### PR DESCRIPTION
**Pull request by azizhk**
A lot of links were broken on the website when trying to open them in a new tab. This is because they were relative urls like `docs/css`, so if you were on a pages like `https://emotion.sh/docs/install` then the url would become `https://emotion.sh/docs/docs/css`. Double `/docs/`
The urls were working if opened in the same tab because the gatsby-plugin-catch-links was somehow catching them correctly.

I ran http://www.brokenlinkcheck.com/ on https://emotion.sh and got the following 404s.
screen shot 2018-02-18 at 5 27 02 pm

My approach was that the links should work on github as well as on the website.
Like if you go to https://github.com/azizhk/emotion/blob/1bf4eba72a8e11660cb4dc7b7a4ed717afae9b58/docs/babel.md and click on the last link then it should work and take you to the corresponding markdown file in the repository.

### CODE REVIEW

File: packages/site/gatsby-node.js [hunk 1]
```diff
@@ -244,6 +244,22 @@ exports.setFieldsOnGraphQLNodeType = ({ type }) => {
...
+                node.properties &&
+                node.properties.href &&
+                !node.properties.href.startsWith('//') &&
+                !node.properties.href.startsWith('http')
+              ) {
+                node.properties.href = node.properties.href
+                  .replace(/\.md(#.*)?$/, (match, hash) => {
+                    return hash
+                  })
+                  .replace(/^\/packages\//, '/docs/')
```
Comment (azizhk) [inline packages/site/gatsby-node.js]: All the main code for creating proper urls is here.

File: packages/emotion/README.md [hunk 2]
```diff
@@ -32,27 +32,26 @@ minimizes the runtime cost of css-in-js dramatically by parsing your styles with
...
-- [Extract Static](https://github.com/emotion-js/emotion/tree/master/docs/extract-static.md)
+- [Extract Static](/docs/extract-static.md)
 
 - API
-  - [css](https://github.com/emotion-js/emotion/tree/master/docs/css.md)
-  - [styled](https://github.com/emotion-js/emotion/tree/master/docs/styled.md)
-  - [css prop](https://github.com/emotion-js/emotion/tree/master/docs/css#CSS-Prop.md)
-  - [composition](https://github.com/emotion-js/emotion/tree/master/docs/composition.md)
-  - [keyframes](https://github.com/emotion-js/emotion/tree/master/docs/keyframes.md)
-  - [fontFace](https://github.com/emotion-js/emotion/tree/master/docs/font-face.md)
```
Comment (azizhk) [inline packages/emotion/README.md]: deleted fontFace, could not find appropriate url

**Comment by azizhk** [comment]
NPM also follows relative links when present in Readme.md of packages.
Example:
https://www.npmjs.com/package/zeroclipboard#other-limitations
https://github.com/zeroclipboard/zeroclipboard/blame/09660421dd377fda5bd66ce9bb69c916470bfbb5/README.md#L52

I've changed only one url from emotion website to relative in babel-plugin-emotion, will revert that to point to the website.

**Comment by mitchellhamilton** [comment]
@azizhk Yep, make that change.

**Comment by azizhk** [comment]
Ran http://www.brokenlinkcheck.com/ on https://deploy-preview-576--emotion.netlify.com/
screen shot 2018-02-20 at 12 25 42 am

**Comment by tkh44** [comment]
Thank you!

**Comment by azizhk** [comment]
Hey yeah sorry about the lint errors, actually I somehow could not setup lerna properly.
I ran `yarn install` and then `yarn bootstrap` but somehow it did not install all dependencies and maybe that is why the githooks were not installed.

I `cd`ed  to `packages/site` and then ran `npm install` to work on the website.
Don't know if anyone else is facing this issue.
Config:
macOS High Sierra
yarn -v: 1.3.2
node -v: v8.9.4
npm -v: 5.6.0

Also please need help with https://github.com/emotion-js/emotion/issues/573

**Comment by mitchellhamilton** [comment]
@azizhk you don't actually need to run yarn bootstrap anymore (we need to update the CONTRIBUTING.md), we don't have any githooks. What do you mean by it didn't install all the dependencies? What happened instead of everything being installed?

**Comment by azizhk** [comment]
So I ran `yarn start:site` and it said it could not find `babel-plugin-emotion`.
I think the linking did not happen.
screen shot 2018-02-20 at 1 30 43 pm

Would you like a PR to add githooks? (Using husky & lint-staged)

**Comment by mitchellhamilton** [comment]
@azizhk You need to run `yarn build` to build a version of babel-plugin-emotion that the site can use.

Personally, I'm not big on githooks because I find they get annoying since I already having linting in my editor but I wouldn't mind hearing @tkh44's opinion, I don't care that much so if @tkh44 thinks it's a good idea then sure.

**Pull request merged**